pkg/scheduler/partition.go

```diff
@@ -324,12 +325,14 @@ func (pc *PartitionContext) AddApplication(app *objects.Application) error {
 		if isRecoveryQueue {
 			queue, err = pc.createRecoveryQueue()
 			if err != nil {
-				return fmt.Errorf("failed to create recovery queue %s for application %s", common.RecoveryQueueFull, appID)
+				outter_err := fmt.Errorf("failed to create recovery queue %s for application %s", common.RecoveryQueueFull, appID)
+				return errors.Join(outter_err, err)
 			}
 		} else {
 			queue, err = pc.createQueue(queueName, app.GetUser())
 			if err != nil {
-				return fmt.Errorf("failed to create rule based queue %s for application %s", queueName, appID)
+				outter_err := fmt.Errorf("failed to create rule based queue %s for application %s", queueName, appID)
+				return errors.Join(outter_err, err)
```

Comment:
ditto: `return errors.Join(fmt.Errorf("failed to create rule based queue %s for application %s", queueName, appID), err)`